src/controllers/actions.ts

```diff
@@ -0,0 +1,51 @@
+// import {getDataLocal, storeDataLocal} from './localStorage';
+
+import {getUserByEmail} from './firebaseFirestore';
+import {getDataLocal, storeDataLocal} from './localStorage';
+
+// export const addContactList = async (uid: string, contactId: string) => {
+//   try {
+//     const oldContactList = await getDataLocal('contactList');
+//     if (oldContactList) {
+//       console.log(oldContactList);
+//       const newList: string[] = oldContactList.push(contactId);
+//       storeData('contactList', newList);
+//       addContact(uid, newList);
+//       console.log(newList);
+//     } else {
+//       storeData('contactList', [uid]);
+//     }
+//   } catch (error) {
+//     console.log(error);
+//   }
+// };
+export const addContact = async (email: string) => {
+  try {
+    const newContact = await getUserByEmail(email);
+    if (newContact) {
+      const oldContactList = await getDataLocal('contactList');
+      if (oldContactList) {
+        const newContactList = [...oldContactList, newContact];
+        await storeDataLocal('contactList', newContactList);
+      } else {
+        await storeDataLocal('contactList', [newContact]);
+      }
+      return newContact;
+    } else {
```

Comment:
one way of simplifying this code since line 27 will be something like this:

```typescript
const oldContactList = await getDataLocal('contactList');
if(!oldContactList){
return null}
await storeDataLocal('contactList', oldContactList ? [...oldContactList, newContact] : [newContact])
return newContact
```

Reply:
Remember that **return** finishes the scope, so else is not required after a return:

for example:
```typescript
if(condition){
return 'blabla'}
else {
return 'bleble'}
```

should be simplified to
``` typescript
if(condition){
return 'blabla'}
return 'bleble'
```

or even more simple:
```typescript
return condition ? 'blabla' : 'bleble'
```